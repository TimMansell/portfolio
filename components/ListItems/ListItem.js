import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'components/Icon';
import Labels from '../Labels';

import { useIsFlexGapSupported } from 'hooks/useFlexGap';

import styles from './ListItem.module.scss';

export const ListItem = ({ item }) => {
  const hasFlexGap = useIsFlexGapSupported();

  const labels = [
    {
      type: 'tertiary',
      items: item.list,
    },
  ];

  const itemClasses = classnames(styles.item, {
    [styles.flexGap]: hasFlexGap,
  });

  return (
    <div className={itemClasses}>
      <div className={styles.icon}>
        <Icon name={[item.icon.family, item.icon.name]} size="sm" />
      </div>
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <Labels labels={labels} type="tertiary" size="lg" centered />
      </div>
    </div>
  );
};

ListItem.propTypes = {
  item: PropTypes.shape({
    icon: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    list: PropTypes.array.isRequired,
  }).isRequired,
};

export default ListItem;
