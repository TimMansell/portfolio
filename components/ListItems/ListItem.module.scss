@import "styles/scssvars";
@import "styles/utilities/pxtorem";

.item {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: var(--vr-sm);

  @media (min-width: $bp-sm) {
    flex-direction: row;
  }
}

.title {
  font-family: var(--font-family-secondary);
  font-size: pxtorem(36);
  letter-spacing: pxtorem(1);
  text-align: center;
  margin-bottom: var(--vr-sm);

  @media (min-width: $bp-sm) {
    text-align: unset;
    font-size: pxtorem(32);
  }
}

.content {
  width: 100%;
}

.icon {
  display: flex;
  justify-content: center;
  align-self: center;
  width: 20%;
  margin-bottom: var(--vr-sm);

  @media (min-width: $bp-sm) {
    align-self: unset;
    margin-right: var(--vr-sm);
  }
}
