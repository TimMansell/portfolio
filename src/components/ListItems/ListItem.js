import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';
import Labels from '../Labels';

import styles from './ListItem.module.scss';

export const ListItem = ({ item }) => {
  return (
    <div className={styles.item}>
      <div className={styles.icon}>
        <Icon name={[item.icon.family, item.icon.name]} size="sm" />
      </div>
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <Labels items={item.list} type="tertiary" size="lg" centered />
      </div>
    </div>
  );
};

ListItem.propTypes = {
  item: PropTypes.shape({
    icon: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    list: PropTypes.array.isRequired,
  }).isRequired,
};

export default ListItem;
