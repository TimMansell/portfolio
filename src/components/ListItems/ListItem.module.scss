@import 'variables';

.item {
  composes: divider from '~scss/helpers/divider.scss';
}

.list {
  list-style-type: none;
  font-size: 20px;
  text-shadow: -1px -1px rgba(0, 0, 0, 0.2);

  @media (min-width: $bp-sm) {
    font-size: 18px;
  }
}

.listitem {
  margin-bottom: 5px;
}

.link {
  composes: linkalt from '~scss/helpers/link.scss';
}
