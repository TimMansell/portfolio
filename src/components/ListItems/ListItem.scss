@import 'variables';

/* stylelint-disable-next-line selector-class-pattern */
.list-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 15px;

  @media (min-width: $bp-sm) {
    margin-bottom: 0;
  }

  &::after {
    content: '';
    margin-top: 20px;
    width: 60%;
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    box-shadow: 0 1px rgba(255, 255, 255, 0.05);

    @media (min-width: $bp-sm) {
      content: none;
    }
  }

  &:last-of-type {
    &::after {
      content: none;
    }
  }

  &__list {
    list-style-type: none;
    font-size: 20px;
    text-shadow: -1px -1px rgba(0, 0, 0, 0.2);

    @media (min-width: $bp-sm) {
      font-size: 18px;
    }
  }

  &__item {
    margin-bottom: 5px;
  }
}
