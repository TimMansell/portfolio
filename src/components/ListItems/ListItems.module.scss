@import 'variables';
@import 'utilities/pxtorem';

.items {
  display: grid;
  grid-gap: var(--vr-md);
  color: var(--secondary-text-colour);

  @media (min-width: $bp-lg) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media (min-width: $bp-xl) {
    grid-template-columns: repeat(3, 1fr);
  }
}
