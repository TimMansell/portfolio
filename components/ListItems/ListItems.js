import React from 'react';
import PropTypes from 'prop-types';

import ListItem from './ListItem';
import styles from './ListItems.module.scss';

export const ListItems = ({ items }) => {
  return (
    <div className={styles.items}>
      {items.map((item, i) => (
        <ListItem item={item} key={i} />
      ))}
    </div>
  );
};

ListItems.propTypes = {
  items: PropTypes.array.isRequired,
};

export default ListItems;
