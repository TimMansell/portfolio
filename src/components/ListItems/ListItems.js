import React from 'react';
import PropTypes from 'prop-types';

import ListItem from './ListItem';
import './ListItems.scss';

export const ListItems = ({ items }) => {
  return (
    <div className="list-items">
      {items.map((item, i) => (
        <ListItem item={item} key={i} />
      ))}
    </div>
  );
};

ListItems.propTypes = {
  items: PropTypes.array.isRequired,
};

export default ListItems;
