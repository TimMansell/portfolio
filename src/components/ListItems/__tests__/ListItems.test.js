import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import ListItems from '../ListItems';

const defaultProps = {
  items: [
    {
      title: 'title 1',
      list: ['list name 1'],
      icon: {
        name: 'css3-alt',
        family: 'fab',
      },
    },
    {
      title: 'title 2',
      list: ['list name 2'],
      icon: {
        name: 'css3-alt',
        family: 'fab',
      },
    },
  ],
};

describe('ListItems', () => {
  it('should render my component', () => {
    render(<ListItems {...defaultProps} />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<ListItems {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should list correct amount of items', () => {
    const { getAllByTestId } = render(<ListItems {...defaultProps} />);

    expect(getAllByTestId('list-item')).toHaveLength(2);
  });
});
