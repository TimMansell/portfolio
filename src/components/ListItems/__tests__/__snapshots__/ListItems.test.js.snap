// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="items"
  >
    <div
      class="item"
      data-testid="list-item"
    >
      <div
        class="icon"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-css3-alt fa-w-12 icon iconSm"
          data-icon="css3-alt"
          data-prefix="fab"
          data-testid="icon"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32l34.9 395.8L192 480l157.1-52.2L384 32H0zm313.1 80l-4.8 47.3L193 208.6l-.3.1h111.5l-12.8 146.6-98.2 28.7-98.8-29.2-6.4-73.9h48.9l3.2 38.3 52.6 13.3 54.7-15.4 3.7-61.6-166.3-.5v-.1l-.2.1-3.6-46.3L193.1 162l6.5-2.7H76.7L70.9 112h242.2z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="content"
      >
        <h3
          class="title"
        >
          title 1
        </h3>
        <ul
          class="labels isLarge isCentered"
          data-testid="labels"
        >
          <li
            class="label isTertiary isLarge"
            data-testid="label"
          >
            list name 1
          </li>
        </ul>
      </div>
    </div>
    <div
      class="item"
      data-testid="list-item"
    >
      <div
        class="icon"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-css3-alt fa-w-12 icon iconSm"
          data-icon="css3-alt"
          data-prefix="fab"
          data-testid="icon"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 32l34.9 395.8L192 480l157.1-52.2L384 32H0zm313.1 80l-4.8 47.3L193 208.6l-.3.1h111.5l-12.8 146.6-98.2 28.7-98.8-29.2-6.4-73.9h48.9l3.2 38.3 52.6 13.3 54.7-15.4 3.7-61.6-166.3-.5v-.1l-.2.1-3.6-46.3L193.1 162l6.5-2.7H76.7L70.9 112h242.2z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="content"
      >
        <h3
          class="title"
        >
          title 2
        </h3>
        <ul
          class="labels isLarge isCentered"
          data-testid="labels"
        >
          <li
            class="label isTertiary isLarge"
            data-testid="label"
          >
            list name 2
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
